python/Ganga/GPIDev/Adapters/IGangaFile.py

```diff
@@ -36,8 +40,44 @@ def location(self):
     def get(self):
         """
         Retrieves locally all files that were uploaded before that 
-        """
-        raise NotImplementedError
+        Order of priority about where a file is going to be placed are:
+            1) The localDir as defined in the schema
+            2) The Job outpudir of the parent job if the localDir is not defined
+            3) raise an exception if neither are defined correctly
+        """
+        if self.localDir:
+            if not os.path.isdir(self.localDir):
+                msg = "Folder '%s' doesn't exist. Please construct this before 'get'-ing a file." % self.localDir
+                logger.error(msg)
+                #os.makedirs(self.localDir)
+                raise GangaException(msg)
```

Comment:
Since this code is usually called by the user, it would be strange for them to see both an error message printed to the log as well as an exception raised.